kubespawner/spawner.py

```diff
@@ -1774,72 +1877,28 @@ def _start(self):
             self._last_event = events[-1]["metadata"]["uid"]
 
         if self.storage_pvc_ensure:
-            # Try and create the pvc. If it succeeds we are good. If
-            # returns a 409 indicating it already exists we are good. If
-            # it returns a 403, indicating potential quota issue we need
-            # to see if pvc already exists before we decide to raise the
-            # error for quota being exceeded. This is because quota is
-            # checked before determining if the PVC needed to be
-            # created.
-
             pvc = self.get_pvc_manifest()
 
-            try:
-                yield self.asynchronize(
-                    self.api.create_namespaced_persistent_volume_claim,
-                    namespace=self.namespace,
-                    body=pvc
-                )
-            except ApiException as e:
-                if e.status == 409:
-                    self.log.info("PVC " + self.pvc_name + " already exists, so did not create new pvc.")
-
-                elif e.status == 403:
-                    t, v, tb = sys.exc_info()
-
-                    try:
-                        yield self.asynchronize(
-                            self.api.read_namespaced_persistent_volume_claim,
-                            name=self.pvc_name,
-                            namespace=self.namespace)
-
-                    except ApiException as e:
-                        raise v.with_traceback(tb)
-
-                    self.log.info("PVC " + self.pvc_name + " already exists, possibly have reached quota though.")
-
-                else:
-                    raise
-
+            # If there's a timeout, just let it propagate
+            yield exponential_backoff(
+                partial(self._make_create_pvc_request, pvc, self.k8s_api_request_timeout),
+                f'Could not create pod {self.pvc_name}',
```

Comment:
Should be something like "Could not create PVC" right? Copy/paste error?

Reply:
Ah! Good catch!